core/src/main/scala/filodb.core/memstore/PartKeyLuceneIndex.scala

```diff
@@ -959,6 +960,48 @@ class PartKeyLuceneIndex(ref: DatasetRef,
     partIdFromPartKeyLookupLatency.record(System.nanoTime - startExecute)
     chosenPartId
   }
+
+  /**
+   * Iterate through the LuceneIndex and calculate cardinality count
+   */
+  def calculateCardinality(partSchema: PartitionSchema, cardTracker: CardinalityTracker): Unit = {
+    val coll = new CardinalityCountBuilder(partSchema, cardTracker)
```

Comment:
Since this builds the cardinality store from scratch, where do we perform the switch of the existing Rocks DB store with a new one?

Reply:
That logic will come up in the upcoming PR. Keeping the scope of this PR simple